core/interface.go

```diff
@@ -12,28 +11,40 @@ import (
 // Advertisement type
 type Advertisement schema.Advertisement
 
+// AdvLink type
+type AdvLink schema.Link_Advertisement
+
+// Index type
+type Index schema.Index
+
+// IndexLink type
+type IndexLink schema.Link_Index
+
 // Interface for a reference provider
 type Interface interface {
 	// PublishLocal provides a new advertisement locally.
-	PublishLocal(ctx context.Context, ad Advertisement) error
+	PublishLocal(ctx context.Context, adv Advertisement) (cid.Cid, error)
```

Comment:
Would it make sense to update the documentation with what the returned CID represents?